lib/jormungandr/src/Cardano/Wallet/Jormungandr/Api/Types.hs

```diff
@@ -75,27 +78,22 @@ import qualified Servant.API.ContentTypes as Servant
 newtype ApiAccountId = ApiAccountId { getApiAccountId :: Bech32.DataPart }
     deriving (Eq, Show)
 
+data AccountState = AccountState
+    { currentBalance
+        :: !(Quantity "lovelace" Word64)
```

Comment:
> AccountState and ApiAccountState, with the only difference being the field names.

Actually, `ApiAccountState` is a more complicated type, because:

1. It requires an additional type (`ApiAccountDelegationState`) to encode the list of delegations. This is because our `Api` types have a one-to-one mapping with the JSON on-the-wire encoding, which enables us to use a generic JSON encoding, and the JSON encoding is also nested.

2. It nests many things in `ApiT`. But for consumers of the type, `ApiT` is just something to unwrap. It has no real purpose outside of the `Api.Types` module.

Have a look at this diff, which shows a piece of client code, when the return type of `getAccountState` was changed from `ApiAccountState` to `AccountState`:

```diff
- eitherDecode testAccountState `shouldBe` Right ApiAccountState
-     { currentBalance =
-         ApiT (Quantity testBalance)
-     , totalTransactionCount =
-         ApiT (Quantity testTransactionCount)
-     , delegationInfo = ApiAccountDelegationInfo
-         { stakePools =
-             [ ( ApiT testPoolId1
-               , ApiT (Quantity testPoolRatio1))
-             , ( ApiT testPoolId2
-               , ApiT (Quantity testPoolRatio2))
-             ]
-         }
+ eitherDecode testAccountState `shouldBe` Right AccountState
+     { currentBalance = Quantity testBalance
+     , totalTransactionCount = Quantity testTransactionCount
+     , stakePools =
+         [ (testPoolId1, Quantity testPoolRatio1)
+         , (testPoolId2, Quantity testPoolRatio2)
+         ]
```